ecl/eclagent/eclgraph.cpp

```diff
@@ -1519,13 +1519,17 @@ void EclAgent::executeThorGraph(const char * graphName)
     WUState state = WUStateUnknown;
     if (agentTopology->hasProp("@queue"))
     {
-        VStringBuffer queueName("%s.thor", agentTopology->queryProp("@queue"));
-        DBGLOG("Queueing wuid=%s, graph=%s, on queue=%s, timelimit=%u seconds", wuid.str(), graphName, queueName.str(), timelimit);
-        Owned<IJobQueue> queue = createJobQueue(queueName.str());
-        queue->connect(false);
-        VStringBuffer jobName("%s/%s", wuid.get(), graphName);
-        IJobQueueItem *item = createJobQueueItem(jobName);
-        queue->enqueue(item);
+        if (executeGraphOnLingeringThor(*queryWorkUnit(), graphName))
+            PROGLOG("Existing lingering Thor handled graph: %s", graphName);
+        else
+        {
+            VStringBuffer queueName("%s.thor", agentTopology->queryProp("@queue"));
+            DBGLOG("Queueing wuid=%s, graph=%s, on queue=%s, timelimit=%u seconds", wuid.str(), graphName, queueName.str(), timelimit);
+            Owned<IJobQueue> queue = createJobQueue(queueName.str());
```

Comment:
Do we need a queue->connect() here ?

Reply:
no, it was there before but it was a mistake, and actually cause a problem on shutdown (that was hidden by container shutdown).
Only need it if wanting to listen to the queue, not post to it as it is in this context.